Examples/Tests/particle_fields_diags/analysis_particle_diags_impl.py

```diff
@@ -0,0 +1,162 @@
+#!/usr/bin/env python3
+
+# Copyright 2019-2021 Luca Fedeli, Yinjian Zhao
+#
+# This file is part of WarpX.
+#
+# License: BSD-3-Clause-LBNL
+
+# This script tests the reduced particle diagnostics.
+# The setup is a uniform plasma with electrons, protons and photons.
+# Various particle and field quantities are written to file using the reduced diagnostics
+# and compared with the corresponding quantities computed from the data in the plotfiles.
+
+import os
+import sys
+
+import numpy as np
+from scipy.constants import c
+from scipy.constants import epsilon_0 as eps0
+from scipy.constants import m_e, m_p
+from scipy.constants import mu_0 as mu0
+import yt
+
+sys.path.insert(1, '../../../../warpx/Regression/Checksum/')
+import checksumAPI
+
+# gamma threshold to switch between the relativistic expression of
+# the kinetic energy and its Taylor expansion.
+gamma_relativistic_threshold = 1.005
+
+def do_analysis(single_precision = False):
+    fn = sys.argv[1]
+
+    ds = yt.load(fn)
+    ad = ds.all_data()
+    ad0 = ds.covering_grid(level=0, left_edge=ds.domain_left_edge, dims=ds.domain_dimensions)
+
+    #--------------------------------------------------------------------------------------------------
+    # Part 1: get results from plotfiles (label '_yt')
+    #--------------------------------------------------------------------------------------------------
+
+    # Quantities computed from plotfiles
+    values_yt = dict()
+
+    domain_size = ds.domain_right_edge.value - ds.domain_left_edge.value
+    dx = domain_size / ds.domain_dimensions
+
+    # Electrons
+    x = ad['electrons', 'particle_position_x'].to_ndarray()
+    y = ad['electrons', 'particle_position_y'].to_ndarray()
+    z = ad['electrons', 'particle_position_z'].to_ndarray()
+    uz = ad['electrons', 'particle_momentum_z'].to_ndarray() / m_e / c
+    w  = ad['electrons', 'particle_weight'].to_ndarray()
+
+    x_ind = np.array([int(xv) for xv in (x - ds.domain_left_edge[0].value) / dx[0]])
```

Comment:
This can be simplified, `x_ind = ((x - ds.domain_left_edge[0].value) / dx[0]).astype(int)`